modules/imgcodecs/test/test_png.cpp

```diff
@@ -7,6 +7,17 @@ namespace opencv_test { namespace {
 
 #if defined(HAVE_PNG) || defined(HAVE_SPNG)
 
+TEST(imRead, passing_Mat)
+{
+  Mat img = Mat::zeros(1000, 1000, CV_8UC3);
+  Mat subImg = img.rowRange(0, 512).colRange(0, 512);
+  imread(samples::findFile("lena.png"), subImg);
+  Mat ori = imread(samples::findFile("lena.png"));
+  EXPECT_EQ(cv::norm(ori, subImg, NORM_INF), 0);
```

Comment:
Signature for `_EQ` is `ASSERT_EQ(expected_reference, actual_result);`.
Correct order of parameters are required to emit valid error messages.

Reference: https://github.com/opencv/opencv/blob/4.0.0/modules/ts/include/opencv2/ts/ts_gtest.h#L8196-L8200

```
GTEST_API_ AssertionResult EqFailure(const char* expected_expression,
                                     const char* actual_expression,
                                     const std::string& expected_value,
                                     const std::string& actual_value,
                                     bool ignoring_case);
```